src/utils.js

```diff
@@ -175,7 +172,76 @@ function taggedString(chaines, ...cles) {
     };
 }
 
+const ALPHANUM = "abcdefghijklmnopqrstuvwxyz0123456789";
+
+function uniqueSlug(size) {
+    let slug = "";
+
+    for (let idx = size; idx--;) {
+        slug += ALPHANUM[(Math.random() * ALPHANUM.lentgth | 0)];
+    }
+
+    return slug;
+}
+
+// eslint-disable-next-line
+const FORBIDDEN_FILENAME_CHAR_REG = /[<>:"\/\\|?*\x00-\x1F]/g;
+
+function filenamify(filename, replacement = "!") {
+    return filename.replace(FORBIDDEN_FILENAME_CHAR_REG, replacement);
+}
+
+// from https://github.com/you-dont-need/You-Dont-Need-Lodash-Underscore#_get
+function get(obj, path, defaultValue) {
+    function travel(regexp) {
+        return String.prototype.split
+            .call(path, regexp)
+            .filter(Boolean)
+            .reduce((res, key) => (res !== null && res !== undefined ? res[key] : res), obj);
+    }
+
+    const result = travel(/[,[\]]+?/) || travel(/[,[\].]+?/);
+
+    return result === undefined || result === obj ? defaultValue : result;
+}
+
+function checkPortAvailability(port) {
+    return new Promise((resolve, reject) => {
+        const tester = net.createServer()
+            .once("error", (error) => {
+                if (error.message.match(/EADDRINUSE/)) {
+                    resolve(false);
+                }
+                else {
+                    reject(error);
+                }
+            })
+            .once("listening", () => {
+                tester.once("close", () => resolve(true)).close();
+            })
+            .listen(port);
+    });
+}
+
+async function getPort(portNumber) {
+    let idx = portNumber;
+
+    while (idx < 65535) {
```

Comment:
Ports below 1024 arent's suitable for node-secure, should sanity check the `portNumber` variable.

Reply:
can't we use port `0` instead and let the operating system choose an available port for us?

Reply:
I didn't even know that we can do that!  
As you want guys, the advantage of this method is that most of the time you can have a fixed time so your user wont get confused when they want to open the web app